airgun/views/ansible_role.py

```diff
@@ -0,0 +1,72 @@
+from widgetastic.widget import Checkbox
+from widgetastic.widget import Table
+from widgetastic.widget import Text
+from widgetastic_patternfly import BreadCrumb
+from widgetastic_patternfly4 import Button
+from widgetastic_patternfly4 import PatternflyTable
+
+from airgun.views.common import BaseLoggedInView
+from airgun.views.common import SearchableViewMixin
+from airgun.widgets import ActionsDropdown
+from airgun.widgets import Pagination
+from airgun.widgets import Pf4ConfirmationDialog
+
+
+class ImportPagination(Pagination):
+    PER_PAGE_BUTTON_DROPDOWN = ".//div[button[@id='pagination-options-menu-toggle-2']]"
+
+
+class AnsibleRolesView(BaseLoggedInView, SearchableViewMixin):
+    """Main Ansible Roles view. Prior to importing any roles, only the import_button
+    is present, without the search widget or table.
+    """
+
+    title = Text("//h1[contains(., text()='Ansible Roles')")
+    import_button = Text("//a[contains(@href, '/ansible_roles/import')]")
+    submit = Button('Submit')
+    total_imported_roles = Text("//span[contains(@class, 'pagination-pf-items-total')]")
+    table = Table(
+        './/table',
+        column_widgets={
+            'Name': Text("./a"),
```

Comment:
I know there's a long history of discussion about this table definition and its behavior.

Are you trying to define column widgets here for every column header? Usually you would only define the column that actually has an interactable widget.

The `Table` widget otherwise has properties/attributes/functions to deal with the table headers/columns, outside of `column_widgets`.

Non-blocking, but its a bit odd to see `Text` widgets being used here, the table cells are just 'normal' text.